### DISCUSSION TITLE
Training to detect closed eyes

### DISCUSSION THREAD

**Discussion by UlrichVonRekkenin** [davisking/dlib#2958] [Q&A]
Hi there,

1. Is it a good idea to use [this one](https://github.com/davisking/dlib/blob/master/examples/train_object_detector.cpp) example or modified version of [another one](https://github.com/davisking/dlib/blob/master/examples/fhog_object_detector_ex.cpp) to train net to detect closed eyes?
2. How to get more then 8*8 fhog image 
![Learned fHOG detector](https://github.com/davisking/dlib/assets/1167164/e1ee3714-09b4-4ebb-9c4b-263be7c2a8ea)
3. Because `train_object_detector.cpp` from the same `training.xml` (produced by `imglab`) generate something like this one 13*11
![detector](https://github.com/davisking/dlib/assets/1167164/a3c26dc4-766f-45c8-8ad7-40c75ce78093)
4. I tried to use frontal close-up images with eyes (strictly horizontal)
5. in the `imglab` mark up left and right eye
6. used flip

When I `dlib::deserialize` and process image with opened eye, them detected. But the closed eyes detected too.

**Comment by davisking** [comment]
HOG isn't going to be super great at telling the difference between closed eyes vs open eyes.  I would use a CNN to do that kind of thing if you want to get something accurate.

**Comment by UlrichVonRekkenin** [comment]
Thanks for your reply!

Something like this CNN, with the final fc<1> to get the probability and `loss_binary_log`?

> ![t18](https://github.com/davisking/dlib/assets/1167164/21ed2e8f-6cc4-4ac5-a746-ad5e437861be)

**Reply by arrufat** [reply]
To train a neural network to detect closed eyes, I'd start with this
http://dlib.net/dnn_mmod_ex.cpp.html

Just make sure to read all comments thoroughly, and label your data consistently.
